assessment/collectors/app_migration_analysis.go

```diff
@@ -240,22 +256,26 @@ func ParseJSONWithRetries(model *genai.GenerativeModel, originalPrompt string, o
 		response = strings.ReplaceAll(response, "\t", "")
 		response = strings.TrimSpace(response)
 
-		var result map[string]interface{}
+		var result map[string]any
 		err := json.Unmarshal([]byte(response), &result)
 		if err == nil {
+			logger.Log.Debug("Parsed response: " + response)
 			return response
 		}
 
-		errMessage := fmt.Sprintf("Error: %v", err)
-		logger.Log.Debug("Received error while parsing: " + errMessage)
+		logger.Log.Debug("Received error while parsing: ", zap.Error(err))
 
-		newPrompt := fmt.Sprintf(promptTemplate, originalPrompt, originalResponse, errMessage)
+		newPrompt := fmt.Sprintf(promptTemplate, err.Error(), response)
 
-		logger.Log.Debug(response)
+		logger.Log.Debug("Json retry Prompt: " + newPrompt)
```

Comment:
use a string formatter for logs where possible

Reply:
When string formatter is used here, the Json structure of the prompt is not maintained. This makes it difficult during debugging.

Reply:
this makes it hard to grep logs and will lead to interspersing of log statements when parallel threads are running.